File: src/waypalm-site/src/main/java/waypalm/site/services/EngineSetup.java
package waypalm.site.services;

import org.springframework.beans.factory.annotation.Value;

import javax.inject.Named;

@Named
public class EngineSetup {
    public static final long startTime = System.currentTimeMillis();

    @Value("${engine.url}")
    private String siteUrl;
    @Value("${engine.context}")
    private String servletContext;
    @Value("${engine.debug}")
    private boolean debugMode;

    public long getStartTime() {
        return startTime;
    }

    public String getSiteUrl() {
        return siteUrl;
    }

    public String getServletContext() {
        return servletContext;
    }

    public boolean isDebugMode() {
        return debugMode;
    }
}

File: src/waypalm-site/src/main/java/waypalm/site/services/EngineSetup.kt
package waypalm.site.services

import javax.inject.Named
import javax.inject.Inject
import org.springframework.beans.factory.annotation.Value

Named
public class EngineSetup [Inject](
        Value("\${engine.url}")
        val siteUrl: String,
        Value("\${engine.context}")
        val servletContext: String,
        Value("\${engine.debug}")
        val debugMode: Boolean
)
{
    val startTime = System.currentTimeMillis()
}